polyfills/fm/client/fmradio.js

```diff
@@ -0,0 +1,421 @@
+// navigator.mozFMRadio polyfill!
+// https://developer.mozilla.org/es/docs/Web/API/Navigator/mozFMRadio
+
+// I have to define the service protocol, and this is as good place as any. The
+// protocol will be as follows:
+// * Each request will have a numeric request id, set by the client
+// * The answers will include the request id, so the client can know to which
+//   request the answer corresponds
+
+// For all the get|set operations:
+//  * Request: { id: requestId,
+//               data: {
+//                 operation: 'set'|'get',
+//                 name: propertyName,
+//                 value: propertyValue (set request only)
+//               }
+//             }
+// * Answer: { id: requestId,
+//             result|error: Whatever }
+
+// For all the invoke operations:
+//  * Request: { id: requestId,
+//               data: {
+//                 operation: methodName,
+//                 params: parameters (optional)
+//               }
+//             }
+// * Answer: Let's assume that the request is handled properly
+
+// For the onpropertychange operation
+//  Request: { id: requestId,
+//             data: {
+//             operation: 'onpropertychange',
+//             handler: desiredPropertyChange,
+//             property: propertyToUpdatedWhenTriggered
+//           },
+// Answer:
+//    { id: requestId,
+//      data: {
+//        result: {
+//          handler: desiredPropertyChange,
+//          propertyValue: newValueOfTheProperty
+//        }
+//      }
+//    }
+
+
+(function(window) {
+
+  function debug(t) {
+    console.log(t);
+  }
+
+  if (window.navigator.mozFMRadio) {
+    // Hmm it's already available... so let's just use it and be done with it
+    return;
+  }
+
+  // Wishful thinking at the moment...
+  const FMRADIO_SERVICE = 'https://fmradioservice.gaiamobile.org';
+
+  // It's nice being monothread...
+  var _currentRequestId = 1;
+
+  var serializeDOMRequest = function(data) {
```

Comment:
Sorry :( I just landed the separation of this, so this should not be here (should not be needed anymore)